packages/common/http/src/fetch.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {HttpBackend, HttpErrorResponse, HttpEvent, HttpEventType, HttpHeaders, HttpRequest, HttpResponse, HttpStatusCode,} from '@angular/common/http';
+import {Injectable, Optional} from '@angular/core';
+import {EMPTY, from, Observable, Observer} from 'rxjs';
+import {catchError} from 'rxjs/operators';
+
+const XSSI_PREFIX = /^\)\]\}',?\n/;
+
+/**
+ * Determine an appropriate URL for the response, by checking either
+ * response url or the X-Request-URL header.
+ */
+function getResponseUrl(response: Response): string|undefined {
+  if (response.url) {
+    return response.url;
+  }
+
+  // stored as lowercase in the map
+  const xRequestUrl = 'X-Request-URL'.toLocaleLowerCase();
+  if (response.headers.has(xRequestUrl)) {
+    return response.headers.get(xRequestUrl) ?? undefined;
+  }
+  return undefined;
```

Comment:
This can be simplified if you change the return type of the function.

```suggestion
return response.headers.get(xRequestUrl) 
```